graphdatascience/procedure_surface/api/graph_sampling_endpoints.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from graphdatascience import Graph
+from graphdatascience.procedure_surface.api.base_result import BaseResult
+
+
+class GraphSamplingEndpoints(ABC):
+    """
+    Abstract base class defining the API for graph sampling algorithms algorithm.
+    """
+
+    @abstractmethod
+    def rwr(
+        self,
+        G: Graph,
+        graph_name: str,
+        start_nodes: Optional[List[int]] = None,
+        restart_probability: Optional[float] = None,
+        sampling_ratio: Optional[float] = None,
+        node_label_stratification: Optional[bool] = None,
+        relationship_weight_property: Optional[str] = None,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+    ) -> GraphSamplingResult:
+        """
+        Computes a set of Random Walks with Restart (RWR) for the given graph and stores the result as a new graph in the catalog.
+
+        This method performs a random walk, beginning from a set of nodes (if provided),
+        where at each step there is a probability to restart back at the original nodes.
+        The result is turned into a new graph induced by the random walks and stored in the catalog.
```

Comment:
is this copied from the GDS manual? I've not reviewed much of the client AV2 work so I'm curious what our strategy is here. We copy the first few lines? Or we author new content?

Reply:
OK I checked and this is novel content. In general, there's no one-paragraph summary to copy from the GDS Manual, so perhaps novel content is generally necessary (although some pages have it).

Then we really need to review it for correctness and consistency with the GDS Manual content, lest we risk confusing both us and our users about the source of truth.